### PR TITLE
Add Github Actions CI

### LINKED ISSUE #46: Move CI to GitHub Actions

### PR DESCRIPTION
**Pull request by siddhant-deepsource**
Fixes #46 

Signed-off-by: siddhant-deepsource <siddhant@deepsource.io>

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Set up container
+        run: |
+          # copy SSH keys to ~/.ssh dir
+          mkdir -p ~/.ssh && touch ~/.ssh/id_rsa
+          # configure git to use ssh
+          git config --global url.git@github.com:.insteadOf https://github.com/
+
+      - name: copy ssh keys
+        run: 'echo "$SSH_KEY" > ~/.ssh/id_rsa && chmod 0600 ~/.ssh/id_rsa'
```
Comment (sourya-deepsource) [inline .github/workflows/CI.yml:31]: What's the SSH key for?

**Review by sourya-deepsource — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yml (hunk 1 above)
Comment (siddhant-deepsource) [reply .github/workflows/CI.yml:31]: Cloning `july` repo for testing

Comment (sourya-deepsource) [reply .github/workflows/CI.yml:31]: It is not a private repo.

Comment (siddhant-deepsource) [reply .github/workflows/CI.yml]: Done

**Review by sourya-deepsource — APPROVED** [review]

**Comment by siddhant-deepsource** [comment]
@sourya-deepsource Will there also be any changes in the cli installation script since the path [here](https://github.com/deepsourcelabs/cli/pull/47/files#diff-f2204989446cadb6e460ae9f44678be54ae8cb68c579ea4968e1bdc5845d60f9R96) has been changed to fetch the codepath from an environment variable.

**Pull request merged**